genesis/streaming.py

```diff
@@ -186,16 +200,8 @@ def _raw_stream(self, timeout):
 					return
 				yield HEARTBEAT_SENTINEL
 			else:
-				# check for errors
-				for msg in msgs:
-					if msg.error() is None:
-						continue
-
-					if msg.error().code() == KafkaError._PARTITION_EOF:
-						continue
-
-					raise Exception(msg.error())
-				yield msgs
+				# unpack messages and check for errors
+				yield [msg for msg in validate_and_process(msgs)]
```

Comment:
May be easier to use:
```
yield list(validate_and_process(msgs))
```

Reply:
I agree, that looks better. I added this in as well.